lib/vanity/adapters/mock_adapter.rb

```diff
@@ -134,17 +132,27 @@ def ab_not_showing(experiment, identity)
       end
 
       def ab_add_participant(experiment, alternative, identity)
-        @experiments[experiment] ||= {}
-        @experiments[experiment][:alternatives] ||= {}
-        alt = @experiments[experiment][:alternatives][alternative] ||= {}
+        alt = alternative(experiment, alternative)
         alt[:participants] ||= Set.new
         alt[:participants] << identity
       end
 
+      def ab_seen(experiment, identity, alternative)
+        if ab_assigned(experiment, identity) == alternative
+          return alternative
```

Comment:
Well, I've updated the implementation and tests to match the other adapters (and remove the explicit return) - if we think the mismatch is too confusing we can address it in another PR perhaps?